pkg/endpoint/endpoint.go

```diff
@@ -1634,6 +1634,79 @@ func (e *Endpoint) APICanModifyConfig(n models.ConfigurationMap) error {
 	return nil
 }
 
+func (e *Endpoint) metadataResolver(ctx context.Context, restoredEndpoint bool, initialLabels labels.Labels, bwm bandwidth.Manager, resolveMetadata MetadataResolverCB, blocking bool) (bool, error) {
+	if !e.K8sNamespaceAndPodNameIsSet() {
+		e.Logger(resolveLabels).Debug("Namespace and Pod are not set")
+		return false, nil
+	}
+
+	// copy the base labels into this local variable
+	// so that we don't override 'baseLabels'.
+	controllerBaseLabels := labels.NewFrom(initialLabels)
+
+	ns, podName := e.GetK8sNamespace(), e.GetK8sPodName()
+
+	pod, cp, identityLabels, info, _, err := resolveMetadata(ns, podName)
+	if err != nil {
+		// If we were unable to fetch the k8s endpoints then
+		// we will mark the endpoint with the init identity.
+		if !restoredEndpoint {
+			// Only mark the endpoint with the 'init' identity if we
+			// received an initial set of labels, this happens
+			// during endpoint creation. During restore the
+			// 'initialLabels' is nil.
```

Comment:
The `During restore the 'initialLabels' is nil` comment might be still correct, but it's now irrelevant here. I'd remove it.